rocketpool-daemon/api/obol/manage-charon-service.go

```diff
@@ -0,0 +1,96 @@
+package obol
+
+import (
+	"fmt"
+	"net/url"
+
+	// "log"
+
+	"os/exec"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/gorilla/mux"
+	batch "github.com/rocket-pool/batch-query"
+	"github.com/rocket-pool/rocketpool-go/v2/rocketpool"
+
+	"github.com/rocket-pool/node-manager-core/api/server"
+	"github.com/rocket-pool/node-manager-core/api/types"
+	"github.com/rocket-pool/smartnode/v2/shared/types/api"
+)
+
+// ===============
+// === Factory ===
+// ===============
+
+type ManageCharonServiceContextFactory struct {
+	handler *ObolHandler
+}
+
+func (f *ManageCharonServiceContextFactory) Create(args url.Values) (*ManageCharonServiceContext, error) {
+	c := &ManageCharonServiceContext{
+		handler: f.handler,
+	}
+	return c, nil
+}
+
+func (f *ManageCharonServiceContextFactory) RegisterRoute(router *mux.Router) {
+	server.RegisterSingleStageRoute[*ManageCharonServiceContext, api.ManageCharonServiceData](
+		router, "obol/create-enr", f, f.handler.logger.Logger, f.handler.serviceProvider.ServiceProvider,
+	)
+}
+
+// ===============
+// === Context ===
+// ===============
+
+type ManageCharonServiceContext struct {
+	handler *ObolHandler
+	rp      *rocketpool.RocketPool
+
+	serviceName string
+	action string
+}
+
+func (c *ManageCharonServiceContext) Initialize() (types.ResponseStatus, error) {
+	sp := c.handler.serviceProvider
+	c.rp = sp.GetRocketPool()
+
+	// Requirements
+	status, err := sp.RequireNodeRegistered(c.handler.ctx)
+	if err != nil {
+		return status, err
+	}
+
+	// Bindings
+
+	return types.ResponseStatus_Success, nil
+}
+
+func (c *ManageCharonServiceContext) GetState(mc *batch.MultiCaller) {
+}
+
+func (c *ManageCharonServiceContext) PrepareData(data *api.ManageCharonServiceData, opts *bind.TransactOpts) (types.ResponseStatus, error) {
+	var cmd *exec.Cmd
+
+	// Choose the appropriate command based on the action
+	switch c.action {

```

Comment:
If we are running it always from docker... wouldn't that serviceName always have to be docker? Why don't we use `docker stop` here instead of `systemctl stop`?